javascript/ql/lib/semmle/javascript/NPM.qll

```diff
@@ -29,7 +29,8 @@ class PackageJson extends JsonObject {
       parentDir.getAChildContainer+() = currentDir and
       pkgNameDiff = currentDir.getAbsolutePath().suffix(parentDir.getAbsolutePath().length()) and
       not exists(pkgNameDiff.indexOf("/node_modules/")) and
-      result = parentPkgName + pkgNameDiff
+      result = parentPkgName + pkgNameDiff and
+      not parentPkg.isPrivate()
```

Comment:
The problem I observed was that `getPackageName()` would have multiple results: it's real name, and then a bogus name derived from an enclosing private `package.json`. If a package is public we clearly only want the public name.

It's possible the answer is just to add this to the second clause of this predicate:
```codeql
not exists(this.getPropStringValue("name"))
```
It's hard to say because I honestly do not understand the motivation for having this second clause in `getPackageName()` in the first place.